dbtool: Don't launch server processes with port 0

AJBats opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->
Addresses #7666 by changing dbtool so it skips over zones that have been assigned port 0 in xidb.

## Steps to test these changes

<!-- Clear and detailed steps to test your changes here -->
1. Default xidb
2. build xi_*
3. Run `py -3 dbtool.py`
4. l (launch server)
5. Observe only a single instance of xi_map is started:

```
Launching D:\Projects\LandSandBoatAJBats\server\xi_connect.exe --log log/connect-server.log
Launching D:\Projects\LandSandBoatAJBats\server\xi_search.exe --log log/search-server.log
Launching D:\Projects\LandSandBoatAJBats\server\xi_world.exe --log log/world-server.log
Launching D:\Projects\LandSandBoatAJBats\server\xi_map.exe --log log/map-server-54230.log --ip 127.0.0.1 --port 54230
```

AJBats committed:
dbtool dont start xi_map for 0 port zones

Changes dbtool so it skips over zones that have been assigned port 0 in xidb.

zach2good commented:
See what I mentioned here: https://github.com/LandSandBoat/server/issues/7666

We don't want a `ports` collection to be passed around and have an invalid port in it. We should be filtering out invalid ports as soon as possible so we can just use `ports` without worry later on

AJBats committed:
Strip ports of invalid 0

Instead of filtering with an if statement we'll strip out 0's from the ports list as soon as possible. Then the rest of the function can proceed as normal.
